### PR TITLE
Theme Support

### PR DESCRIPTION
**Pull request by zachstence**
### Description

Adds support for a dark theme, and customization of light/dark themes

Does not include any work to replace hardcoded colors (e.g. `text-green-500`) with semantic color tokens (e.g. `text-success`).

https://github.com/user-attachments/assets/89e8320e-1334-4358-99b6-6457e3e41d93

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable
- ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

### CODE REVIEW

**Review by archiewood — APPROVED** [review]
works with both system wide and browser-level dark / light / system toggles 👍🏼 

https://github.com/user-attachments/assets/2ba28d95-f437-4111-8f9f-365108f948dd


CleanShot 2024-09-17 at 21 18 35

Something about this seems like the text is misaligned. I think the System either needs to be exactly the same text size, or else a size down

**Comment by zachstence** [comment]
> This looks good me. Do we need tests for any of this?

I created `e2e/themes` which tests the whole flow. I feel there isn't much logic to unit test, I think the E2E test covers it pretty well

File: packages/ui/core-components/src/lib/organisms/layout/EvidenceDefaultLayout.svelte [hunk 1]
```diff
@@ -97,6 +99,9 @@
 			navigator.serviceWorker.register('/service-worker.js');
 		});
 	});
+
+	// TODO where should this go? How do we get project splash to be rendered with the proper theme?
```
[the remark this answers is not in the document]
Comment (zachstence) [reply packages/ui/core-components/src/lib/organisms/layout/EvidenceDefaultLayout.svelte:103]: I'll do that in a different PR, thanks for reminding me

**Comment by ItsMeBrianD** [comment]
:shipit:

**Pull request merged**